addons/portal/controllers/portal.py

```diff
@@ -155,55 +154,116 @@ def counters(self, counters, **kw):
     @route(['/my', '/my/home'], type='http', auth="user", website=True)
     def home(self, **kw):
         values = self._prepare_portal_layout_values()
+        values.update({
+            'partner': request.env.user.partner_id
+        })
         return request.render("portal.portal_my_home", values)
 
     @route(['/my/account'], type='http', auth='user', website=True)
     def account(self, redirect=None, **post):
         values = self._prepare_portal_layout_values()
+
+        partner = request.env.user.partner_id
+        invoice_addresses = partner.child_ids.filtered(lambda address: address.type == 'invoice')
+        delivery_addresses = partner.child_ids.filtered(lambda address: address.type == 'delivery')
+
+        values.update({
+            'page_name': 'my_details',
+            'partner': partner,
+            'invoice_addresses': invoice_addresses,
+            'delivery_addresses': delivery_addresses,
+        })
+        return request.render("portal.portal_my_details", values)
+
+    @route(['/my/address', '/my/address/<int:partner_id>'], type='http', auth='user', website=True, methods=['POST', 'GET'])
+    def address(self, partner_id=None, redirect=None, **post):
+        if request.httprequest.method == 'GET':
+            action = "read" if partner_id else "create"
+        else:
+            action = post.pop('address_action')
+        values = self._prepare_portal_layout_values()
+        Partner = request.env['res.partner'].with_context(show_address=1).sudo()
+        contact = Partner.browse(partner_id) if partner_id else False
```

Comment:
Pretty sure one would be able to sudo remove any partner using this route and passing an arbitrary partner_id parameter.
Make sure to pay double attention to security when you use sudo, especially when the method is exposed publicly like here.